src/operations/scripts.rs

```diff
@@ -0,0 +1,200 @@
+) -> Result<(), Error> {
+    match mode {
+        // Note: we assume it is "safe" to change the target of an existing
+        // symlink
+        Safe => delete_if_link(dest_path),
+        // Note: we need to delete dest_path for unix::fs::symlink to work
+        // later on
+        Override => delete_if_exists(dest_path),
+    }?;
+    println!(
+        "{} -> {}",
+        dest_path.to_string_lossy(),
+        src_path.to_string_lossy()
+    );
+    std::os::unix::fs::symlink(src_path, dest_path).map_err(|e| {
+        new_error(&format!(
+            "Could not create link from {:?} to {:?}: {}",
+            dest_path, src_path, e
+        ))
+    })
+}
+
+#[cfg(unix)]
+fn delete_if_link(path: &PathBuf) -> Result<(), Error> {
+    // This will make an error if the path does not exist,
+    // but also if we don't have permission to read the path.
+    // In both cases, we want to return early - in the later,
+    // the code will probably fail when creating the symlink
+    // later on.
+    let meta = std::fs::symlink_metadata(path);
+    if meta.is_err() {
+        return Ok(());
+    };
+    let meta = meta.unwrap();
+    if !meta.file_type().is_symlink() {
+        return Err(new_error(&format!(
+            "{:?} exists and is *not* a symlink",
+            path
+        )));
+    };
+    std::fs::remove_file(path).map_err(|e| {
+        new_error(&format!(
+            "Could not remove existing symlink {:?}: {}",
+            path, e
+        ))
+    })?;
+    Ok(())
+}
+
+#[cfg(unix)]
+fn delete_if_exists(path: &PathBuf) -> Result<(), Error> {
+    if path.exists() {
+        std::fs::remove_file(path)
+            .map_err(|e| new_error(&format!("Could not remove existing file {:?}: {}", path, e)))?;
+    }
+    Ok(())
+}
+
+fn find_egg_info(project_path: &PathBuf) -> Result<PathBuf, Error> {
+    let pattern = format!("{}/*.egg-info", project_path.to_string_lossy());
+    let glob = glob::glob(&pattern).expect("could not parse glob pattern");
+    let mut matches = vec![];
+    for entry in glob {
+        if let Ok(path) = entry {
+            matches.push(path)
+        }
+    }
+    let num_matches = matches.len();
+    if num_matches != 1 {
+        return Err(new_error(&format!(
+            "Expecting exactly one .egg-info entry, got {}",
+            num_matches
+        )));
+    }
+    Ok(matches[0].clone())
+}
+
+fn read_entry_points(egg_info_path: &PathBuf) -> Result<Vec<String>, Error> {
+    let entry_points_txt_path = egg_info_path.join("entry_points.txt");
+    let config = Ini::load_from_file(&entry_points_txt_path).map_err(|e| {
+        new_error(&format!(
+            "Could not read {:?}: {}",
+            &entry_points_txt_path, e
+        ))
+    })?;
+    let mut res = vec![];
+    let section = config.section(Some("console_scripts"));
+    if let Some(section) = section {
+        for (key, _value) in section.iter() {
+            res.push(key.to_string());
+        }
+    }
+    Ok(res)
```

Comment:
I just felt like golfing so that's probably too compact for comfort =], but I'd at least replace `HashMap::iter` by `HashMap::keys` since you don't use the values at all.
```
Ok(match config.section(Some("console_scripts")) {
    Some(section) => section.keys().map(String::to_owned).collect(),
    None => Vec::new(),
})
```